File: 31st October 2022/C++/Satisfy the equation.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
class Solution {
      public:
    vector<int> satisfyEqn(int a[], int n) {
        // code here
        unordered_map<int,pair<int,int>> m;
        vector<int> Ans;
        for(int i=0;i<n-1;i++){
            for(int j=i+1;j<n;j++){
                int temp=a[i]+a[j];
                if(m.find(temp)==m.end()){
                    m[temp]=make_pair(i,j);
                }
                else{
                    pair<int,int> pp=m[temp];
                    if(pp.first!=i && pp.first!=j && pp.second!=i && pp.second!=j){
                        vector<int> ans(4);
                        ans[0]=pp.first;
                        ans[1]=pp.second;
                        ans[2]=i;
                        ans[3]=j;
                        if(Ans.size()==0)
                            Ans=ans;
                        else{
                            bool  rep=false;
                            for(int i1=0;i1<Ans.size();i1++){
                                if(Ans[i1]<ans[i1])
                                    break;
                                if(Ans[i1]>ans[i1]){
                                    rep=true;
                                    break;
                                }
                            }
                            if(rep)
                                Ans=ans;
                        }
                    }
                }
            }
        }
        if(Ans.size()==0)
            return vector<int>(4,-1);
        else
            return Ans;
    }
};

//{ Driver Code Starts.
int main() {
    int t;
    cin >> t;
    while (t--) {
        int N;
        
        cin>>N;
        int A[N];
        
        for(int i=0; i<N; i++)
            cin>>A[i];

        Solution ob;
        vector<int> ptr = ob.satisfyEqn(A,N);
        
        cout<<ptr[0]<<" "<<ptr[1]<<" "<<ptr[2]<<" "<<ptr[3]<<endl;
    }
    return 0;
}
// } Driver Code Ends

File: Problems & Solutions/S/Satisfy the equation/31st October 2022/Java/Satisfy the equation.java
//{ Driver Code Starts
//Initial Template for Java

import java.io.*;
import java.util.*;

class GFG {
    public static void main(String args[]) throws IOException {
        BufferedReader read =
            new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine());
        while (t-- > 0) {
            int N = Integer.parseInt(read.readLine());
            
            String S[] = read.readLine().split(" ");
            int[] A = new int[N];
            
            for(int i=0; i<N; i++)
                A[i] = Integer.parseInt(S[i]);

            Solution ob = new Solution();
            int[] ptr = ob.satisfyEqn(A,N);
            
            System.out.println(ptr[0] + " " + ptr[1] + " " + ptr[2] + " " + ptr[3]);
        }
    }
}
// } Driver Code Ends



//User function Template for Java

class Solution {
    static int[] satisfyEqn(int[] A, int N) {
        // code here
        int ans[]=new int[4];
        for(int i=0;i<N;i++)
        {
            for(int j=0;j<N;j++)
            {
                for(int k=0;k<N;k++)
                {
                    for(int l=0;l<N;l++)
                    {
                        if(A[i]+A[j]==A[k]+A[l] && i!=j && k!=l && j!=k && i!=k && j!=l && i
                        !=l)
                        {
                            ans[0]=i;ans[1]=j;ans[2]=k;ans[3]=l;
                            return ans;
                        }
                    }
                }
            }
        }
        ans[0]=-1;ans[1]=-1;ans[2]=-1;ans[3]=-1;
        return ans;
    }
};